### ISSUE TITLE
Help set this up in NeoVim using LazyVim

### ISSUE THREAD

**Issue by maulik13** [kcl-lang/vim-kcl#17]
## General Question

I am trying to set up NeoVim for `kcl` and I am not able to make it work using LazyVim. I could get the formatting to work, but I do not see any syntax highlighting. 

Here is my configuration added in a file under plugins.

```lua
vim.filetype.add({
  extension = {
    kcl = "kcl",
  },
})
return {
  {
    "kcl-lang/kcl.nvim",
  },
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        kcl = {},
      },
    },
    setup = {},
  },
  {
    "williamboman/mason.nvim",
    opts = { ensure_installed = { "kcl" } },
  },
}

```

Here is the output from the `LspInfo` command. 

```
- Detected filetype: `kcl`
- 2 client(s) attached to this buffer
- Client: `kcl` (id: 1, bufnr: [2])
  root directory:    ~/work/playground/crossplane-kcl/
  filetypes:         kcl
  cmd:               ~/.local/share/nvim/mason/bin/kcl-language-server
  version:           `kcl-language-server Version: 0.10.8-c020ab3eb4b9179219d6837a57f5d323`
  executable:        true
  autostart:         true
```

Neovim: 0.10.1
OS: OSX Sonoma 14.5
Terminal: Kitty

**Comment by Peefy** [comment]
cc @He1pa

**Comment by maulik13** [comment]
Also, is there a plan for adding `kcl` to treesitter https://github.com/nvim-treesitter/nvim-treesitter?tab=readme-ov-file#supported-languages? It would be really helpful :)

**Comment by Peefy** [comment]
Yes, https://github.com/kcl-lang/tree-sitter-kcl is an ongoing work, and once it has been fully tested we will push it upstream. 😄 cc @He1pa

**Comment by He1pa** [comment]
Is it work for you?
```
require('lazy').setup({
  "kcl-lang/kcl.nvim",
})
```

**Comment by maulik13** [comment]
I tried just using that, but highlights still do not work. It is formatting the code though.

**Comment by He1pa** [comment]
Are other functions of lsp normal, such as goto definition and completion?

**Comment by maulik13** [comment]
Sorry for the late reply, had a break form kcl for a bit. No it seems goto definition does not work either. As for the code completion, I am a bit unsure if NeoVim is using tokens or the actual lsp to list items for auto completion.

**Comment by maulik13** [comment]
I have made some progress. I see that `.k` extension was linked to another filetype. I have now added the following and I see that correct lsp is attached to my buffer. 

```
vim.filetype.add({
  extension = {
    k = "kcl",
  },
})

```

With this, go to definition and formatting is working correctly.

**Comment by He1pa** [comment]
So, this issue has been resolved?

**Comment by maulik13** [comment]
We could close this case yes. Though it would be nice to document that there is a need to override filetype since neovim ships with a default mapping of extension `.k` to `kimwitu++`.
